src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/EnvelopeEventProcessorTest.java

```diff
@@ -122,38 +137,80 @@ public void should_dead_letter_the_message_when_unrecoverable_failure() throws E
     }
 
     @Test
-    public void should_not_finalize_the_message_when_recoverable_failure() throws Exception {
+    public void should_dead_letter_the_message_when_notification_sending_fails() throws Exception {
         // given
-        reset(eventPublisherContainer);
-        willThrow(new RuntimeException("test exception"))
-            .given(eventPublisherContainer)
-            .getPublisher(any(), any());
+        String exceptionMessage = "test exception";
+
+        willThrow(new NotificationSendingException(exceptionMessage, null))
+            .given(processedEnvelopeNotifier)
+            .notify(any());
+
 
         // when
         CompletableFuture<Void> result = processor.onMessageAsync(someMessage);
         result.join();
 
         // then
+        verify(messageOperations).deadLetter(
+            eq(someMessage.getLockToken()),
+            eq("Message processing error"),
```

Comment:
is there a constant for this?

Reply:
I've just created a constant in this test class. In application code it's still one instance of a string (there are no other reasons).